### PR TITLE
Core/Entities: fix Anub'arak ice fear falling through ground

### PR DESCRIPTION
**Pull request by donaldjansen**
**Changes proposed:**

-  This fixes a problem that causes the ice fear to fall through the ground in the last boss anub'arak

**Target branch(es):** 3.3.5

- 3.3.5

**Issues addressed:** Closes # 
There is no issue number for this, it was just a bug I found when we played

**Tests performed:**
I have build the source and tested anub'arak again the ice fear does not fall through ground anymore

### CODE REVIEW

File: src/server/game/Entities/Creature/Creature.cpp [hunk 1]
```diff
@@ -2930,7 +2930,7 @@ void Creature::UpdateMovementFlags()
         return;
 
     // Set the movement flags if the creature is in that mode. (Only fly if actually in air, only swim if in water, etc)
-    float ground = GetFloorZ();
+	float ground = GetMap()->GetHeight(GetPhaseMask(), GetPositionX(), GetPositionY(), GetPositionZMinusOffset());
```
Comment (Aokromes) [inline src/server/game/Entities/Creature/Creature.cpp]: convert tab to space.

Comment (donaldjansen) [inline src/server/game/Entities/Creature/Creature.cpp]: sure will do

Comment (donaldjansen) [inline src/server/game/Entities/Creature/Creature.cpp]: wierd, just checked the code it doesn't seem to have a tab

Comment (donaldjansen) [inline src/server/game/Entities/Creature/Creature.cpp]: Ok I see, VS2017 and Cntrl + K + D = Autoformat with resharper which caused the tab

**Comment by tkrokli** [comment]
This PR is a partial revert of commit https://github.com/TrinityCore/TrinityCore/commit/f6c849729b27b77228704b595de3adaf24da2c10 , specifically `Creature.cpp`: https://github.com/TrinityCore/TrinityCore/commit/f6c849729b27b77228704b595de3adaf24da2c10#diff-f3f953a63e2a008d22de69cb68f735c4 - @Treeston may be interested to see it.

**Comment by Shauren** [comment]
This will not pass as you are undoing performance optimization
Also you are making active gameobjects update large areas of map - I can't see how is this even related to this issue

**Pull request closed without merging**